cpp/rrtmgp_conversion.h

```diff
@@ -914,7 +914,10 @@ struct MemPoolSingleton
   static inline
   void print_state()
   {
-    std::cout << "Used " << s_curr_used << " of out of " << s_mem.size() << ", high_water was " << s_high_water << std::endl;
+    if (s_curr_used != 0) {
```

Comment:
I think the idea was not just to print the fact that "hey, something is still in use!", but also to get an idea of the high watermark for mem usage.

Imho, the best way would be to pass a bool `verbose` to rrtmgp's finalize (default to true, for bwd compatibility) and then call print_state only if `verbose==true`. Then, host code can call finalize choosing independently whether to print or not (e.g., passing `my_rank==0` as `verbose`).

Reply:
Luca is correct. If our high water is only 10% of the total pool, we want to know about it because we are asking for way too much.

Reply:
If you don't want to print on root only, you can still make the finalize signature as `void finalize(bool print_stats = true)`. Then, when called from eamxx, we can pass `false` unless the user asked rrtmgp to print the stats (e.g., we could allow something like `atmchange rrtmgp::finalize_print_stats` or something). We can decide how to handle the "verbose" flag from the app side.

Reply:
Another idea: make finalize and print_state accept an `std::ostream` (which can default to std::cout). The custom app can pass a custom stringstream, in case they want to parse the stats themselves. E.g.:
```c++
void finalize(std::ostream& out) {
  print_stats(out);
  ...
}
```
and from eamxx:
```c++
std::stringstream ss
rrtmgp::finalize(ss);
if (debug) {
  // We can print ss.str(), or even parse it, to extract the high_water number, and compare across ranks
}
```

Reply:
In general, I find it very "polite" when a pkg/lib offers to print to a user-provided buffer rather than autonomously deciding to print to screen, and picks the screen only if not otherwise specified.

Reply:
I think the idea is that all the ranks should have roughly the same high water as root.

Reply:
They are more common in flood prone areas, rivers and coasts.